### PR TITLE
Add git workflow document

### PR DESCRIPTION
**Pull request by derberg**
This PR in a result of a discussion from https://github.com/asyncapi/asyncapi/issues/312 and the SIG open meeting.

Git workflow doc explains that we all (maintainers too) work on forks and how to setup fork to work like a boss so existing maintainers can quickly get into the flow.

I also created a script for setting up a fork, as an alternative to manual steps. Created document links here - https://gist.github.com/derberg/87319e9c486e4a6c9bef5b629ab0d386

Document heavily inspired by the awesome work of Kyma's technical writers - https://kyma-project.io/community/contributing#git-workflow-git-workflow

### CODE REVIEW

File: documents/git-workflow.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+* Create branches on the fork and avoid working directly on the `master` branch.
+
+## Prepare the fork
+
+A fork is a copy of the repository from which you raise pull requests to propose changes to the original repository.
+The unified AsyncAPI contribution workflow that bases on forks allows both the members of the Kyma organization and the external contributors to contribute code and content through the same process. This keeps the main repositories branches clean as contributors create branches only on the forked repositories.
+
+## Configure your fork
+
+The document refers to the original repository as the upstream repository and to the forked repository as the origin repository. We assume you already have a fork of the upstream and you `git clone` it laready.
```
Comment (jonaslagoni) [inline documents/git-workflow.md]:
```suggestion
The document refers to the original repository as the upstream repository and to the forked repository as the origin repository. We assume you already have a fork of the upstream and you `git clone` it already.
```
I mean, `laready` could be a new thing :thinking: :tada:

Comment (derberg) [reply documents/git-workflow.md]: 😘

**Comment by derberg** [comment]
@jonaslagoni well done 🦅 👁 
I commit your suggestion, take a look again

File: documents/git-workflow.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Each contributor and maintainer in AsyncAPI must follow this workflow:
+
+* Work on forked repositories.
+* Create branches on the fork and avoid working directly on the `master` branch.
+
+## Prepare the fork
+
+A fork is a copy of the repository from which you raise pull requests to propose changes to the original repository.
+The unified AsyncAPI contribution workflow that bases on forks allows both the members of the Kyma organization and the external contributors to contribute code and content through the same process. This keeps the main repositories branches clean as contributors create branches only on the forked repositories.
```
Comment (jonaslagoni) [inline documents/git-workflow.md]:
```suggestion
The unified AsyncAPI contribution workflow that bases on forks allows both the members of the AsyncAPI organization and the external contributors to contribute code and content through the same process. This keeps the main repositories branches clean as contributors create branches only on the forked repositories.
```
Guess this is a copy paste error :smile:

Comment (derberg) [reply documents/git-workflow.md]: yeap, huge part of it, thanks
honestly mentioned in PR description :)

Comment (jonaslagoni) [reply documents/git-workflow.md]: Yea I know :+1: No need to reinvent the wheel twice :smile:

**Comment by derberg** [comment]
@jonaslagoni 

you did manual setup or with the script?

**Comment by jonaslagoni** [comment]
I had a rep cloned already so changed the origin to my fork and followed along your guide manually  👍

**Comment by derberg** [comment]
@jonaslagoni you think I can merge? or are we waiting for 🐔@fmvilas🐔 to see if he is brave enough to make a first PR from a fork

**Comment by fmvilas** [comment]
hahaha! I see you haven't noticed but I already did: https://github.com/asyncapi/website/pull/57.

**Comment by derberg** [comment]
It is actually not included in the release. This was an unintentional release caused by the release automation initial run. 1.0 version of the spec was released a long time ago. The release mentioned by the bot was removed.

**Pull request merged**